### DISCUSSION TITLE
How to contribute

### DISCUSSION THREAD

**Discussion by pquerner** [OpenMage/magento-lts#3304] [Questions & Answers] [answered]
Hello!

I've got something to contribute and wanted to create a PR, however I am stuck on this:

```
git pull upstream 1.9.4.x

fatal: couldn't find remote ref 1.9.4.x
```

as mentioned here: 
https://github.com/OpenMage/magento-lts#git

Should I just add my changes to `main` only since 19.x went basically into "patch only" mode?

Also:
What is the preferred way to contribute; first create a issue (or search for similar), wait for response (is that change needed / wanted in OpenMage?); and then finally add a PR; -- or simply drop the PR and discuss in the PR.

Thanks!

**Comment by fballiano** [comment, marked as answer]
I think it should be `git pull upstream main`, I've just created https://github.com/OpenMage/magento-lts/pull/3306 to fix it

**Reply by pquerner** [reply]
`git pull upstream main` worked - so I assume I just target that branch and if needed be it, the code will be merged / backported to `^19` aswell by you guys?

(Just a general question, my intended contribution should work on both 20 and 19)

**Reply by fballiano** [reply]
it will be backported to v19 only if it's a vulnerability or a very important fix :-)

**Reply by pquerner** [reply]
Okay, thanks! :)
